prosody-plugins/mod_auth_token.lua

```diff
@@ -80,6 +80,9 @@ function provider.get_sasl_handler(session)
 		-- here we check if 'room' claim exists
 		local room, roomErr = token_util.get_room_name(token, appSecret);
 		if room == nil then
+            if roomErr == nil then
```

Comment:
When we reach this point, the room is nil, so we check if there was any error during token verification. If the error is nil it means that the claim is missing.